olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -372,6 +373,11 @@ HttpResponse OlpClient::CallApi(std::string path, std::string method,
                                 OlpClient::RequestBodyType post_body,
                                 std::string content_type,
                                 CancellationContext context) const {
+  if (!settings_.network_request_handler) {
+    return HttpResponse(olp::http::HttpStatusCode::SERVICE_UNAVAILABLE,
```

Comment:
`SERVICE_UNAVAILABLE` is a retriable error code. But here it seems the issue is more on the user side. So probably you should use `ErrorCode::OFFLINE_ERROR` instead.